### PR TITLE
feat(x/ecocredit): add BurnRegen method

### PR DESCRIPTION
**Pull request by aaronc**
## Description

This PR adds a `BurnRegen` method to the ecocredit module which allows a user to burn REGEN from their balance with an optional burn reason which can be used by clients to associate the burn with specific activities.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by JeancarloBarrios** [comment]
It looks good; I was just wondering. Does it make sense that it is its own cosmos SDK x module? that could eventually evolve into a fee management module? However, if so, it makes sense to do it after and migrate it to it then.

**Comment by aaronc** [comment]
Thanks for the review @JeancarloBarrios. I agree that a standalone module would make more sense. But also, there's a fair amount of overhead in creating a new module, so I think we will get this live on chain sooner by putting it in x/ecocredit. Also, the other burning functionality we've discussed would be native to x/ecocredit so it's not totally out of place.

**Review by JeancarloBarrios — APPROVED** [review]

**Comment by aaronc** [comment]
It would be great to get one more review here if anyone has time.

**Review by blushi — APPROVED** [review]
LGTM

File: .golangci.yml [hunk 1]
```diff
@@ -5,7 +5,6 @@ linters:
   disable-all: true
   enable:
     - bodyclose
-    - depguard
```
Comment (aaronc) [inline .golangci.yml:8]: This linter was causing CI errors but we never configured it and probably it just got enabled with a default blocking rule in #2148. cc @robert-zaremba

**Pull request merged**